module/Activity/src/Activity/Mapper/Activity.php

```diff
@@ -94,6 +94,118 @@ public function getUpcomingActivities($count = null, $organ = null, $category =
         return $qb->getQuery()->getResult();
     }
 
+    /**
+     * Get upcoming activities sorted by date for member
+     *
+     * @param \User\Model\User $user Option user that should relate to activity
+     *
+     * @return array
+     */
+    public function getUpcomingActivitiesForMember($user)
+    {
+        // Get subscriptions (not including non-approved)
+        $result = $this->getUpcomingActivitiesSubscribedBy($user);
+
+        // Get created by member (including non-approved)
+        $result = array_merge($result, $this->getUpcomingActivitiesCreatedBy($user));
+
+        // Get associated with organs (including non-approved)
+        foreach ($user->getMember()->getCurrentOrganInstallations() as $organ) {
+            $result = array_merge($result, $this->getUpcomingActivitiesByOrgan($organ));
+        }
+
+        // Do sorting based on start time
+        $result = $this->sortMultiArrayByKey($result, 'getBeginTime');
```

Comment:
No need to write a custom sort function, you can use `usort`, something like this should work:

```suggestion
        $result = usort($result, function($a, $b) {
            return $a->getBeginTime() < $b->getBeginTime() ? -1 : 1;
        });
```

Reply:
Thanks for the tip!